sign/ed25519/ed25519.go

```diff
@@ -52,15 +68,31 @@ func (kp *KeyPair) Seed() []byte { return kp.GetPrivate() }
 func (kp *KeyPair) Public() crypto.PublicKey { return kp.GetPublic() }
 
 // Sign creates a signature of a message given a key pair.
-// Ed25519 performs two passes over messages to be signed and therefore cannot
-// handle pre-hashed messages.
+// This function supports all the three signature variants defined in RFC-8032,
+// namely Ed25519 (or pure EdDSA), Ed25519Ph, and Ed25519Ctx.
 // The opts.HashFunc() must return zero to indicate the message hasn't been
 // hashed. This can be achieved by passing crypto.Hash(0) as the value for opts.
+// The opts.HashFunc() must return SHA512 to indicate that the message will be
+// hashed with SHA512. This can be achieved by passing crypto.SHA512 as the value
+// for opts.
```

Comment:
```suggestion
// The opts.HashFunc() must return zero to specify either Ed25519 or Ed25519Ctx 
// variant. This can be achieved by passing crypto.Hash(0) as the value for opts.
// The opts.HashFunc() must return SHA512 to specify the Ed25519Ph variant.
// This can be achieved by passing crypto.SHA512 as the value for opts.
// Use an Options struct to pass a context string for signing.
```